ush/ufsda/misc_utils.py

```diff
@@ -9,6 +11,23 @@
 }
 
 
+def calc_fcst_steps(fcst_step, win_length):
+    """
+    function to return a list of forecast steps
+    for a given fcst_step (forecast step)
+    and win_length (window length)
+    """
+    # need to get +- half of the window length
+    # assumes only hours for now, probably bad...
+    # also assumes the window is symmetric, probably also bad
+    h = int(re.findall('PT(\\d+)H', win_length)[0])
+    start = f'PT{h-h//2}H'
+    end = f'PT{h+h//2}H'
```

Comment:
Sorry about the late reply on this. I tested the newest version of GDASApp this morning and I found additional issue stemming from here. If both `fcst_step` and `win_length` are `'PT6H'`, the return value of this function is `['PT3H', 'PT9H']` but what we really want is `['PT6H']`. So I suggest you add a condition before l24 (something like below):
```
h2 = int(re.findall('PT(\\d+)H', fcst_step)[0])
if h2 > h//2: return [fcst_step]
```

Reply:
@RussTreadon-NOAA I'm debugging a slightly different segfault with GDASApp hofx and 3dvar applications as well. I suspect something wrong with R2D2 because I tried other background files Cory provided months ago with the newest version of fv3-bundle and had no problem running `fv3jedi_hofx_nomodel.x`. If you are interested, here is my directory on Orion so you can experiment with my yaml: `/work/noaa/da/fhan/JEDI/test_humidity_assign_error/`. I used the most up to date `fv3jedi_hofx_nomodel.x` I built this morning. 

I will continue to debug what is causing problems in GDASApp.

Reply:
Thanks @FanHan-NOAA .  Seems adding the line

`
field metadata override: /work/noaa/da/fhan/JEDI/src/fv3-bundle/build/fv3-jedi/test/Data/fieldmetadata/gfs-restart.yaml
`

to my yaml gets me past the stc error.   The 3dvar job later fails when it attempts to read the satbias_cov file.  This file does not exist in the R2D2 database.   I'll update PR #58 since PR #58 deals with sample yaml files.